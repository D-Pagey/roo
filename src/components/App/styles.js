import styled from 'styled-components';

export const Body = styled.p`
    margin: 2.5rem 0;
`;
