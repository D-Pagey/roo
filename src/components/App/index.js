import React, { useEffect, useState } from 'react';
import data from '../../restaurants';
import Navbar from '../Navbar';
import Location from '../Location';
import * as S from './styles';

const App = () => {
    const [neighborhood, setNeighborhood] = useState('');

    useEffect(() => {
        setNeighborhood(data.neighborhood);
    }, []);

    return (
        <div>
            <Navbar />
            <Location location={neighborhood} />
            <S.Body>301 Locations</S.Body>
        </div>
    );
};

export default App;
