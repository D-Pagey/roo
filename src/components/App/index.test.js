import React from 'react';
import { render } from '@testing-library/react';
import { useMediaQuery } from 'react-responsive';
import data from '../../restaurants';
import App from '.';

jest.mock('react-responsive', () => ({
    useMediaQuery: jest.fn(() => false)
}));

describe('App component', () => {
    it('should render', () => {
        const { container } = render(<App />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('shows quantity of restaurants on non-mobile', () => {
        useMediaQuery.mockImplementation(() => true);

        const { getByText } = render(<App />);
        getByText(`${data.restaurants.length} Restaurants`);
    });
});
