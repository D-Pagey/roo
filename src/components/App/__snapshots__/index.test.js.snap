// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 16px;
  height: 48px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: #ffffff;
  border-radius: 3px;
  border: 0;
  color: #00a396;
  cursor: pointer;
  font-size: 1rem;
  padding: 11px 1rem;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #00ccbc;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 177px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.5rem 1rem;
}

.c3 {
  font-size: 1rem;
  line-height: 1.375rem;
  margin: 0 0 2px;
  opacity: 50%;
}

.c4 {
  color: #ffffff;
  font-size: 2.125rem;
  margin: 0 0 1.25rem;
}

.c6 {
  margin: 2.5rem 0;
}

<div>
  <div
    class="c0"
  >
    <img
      alt="deliveroo logo"
      data-testid="navbarSmallLogo"
      src="small-logo.svg"
    />
    <div
      class="c1"
    >
      <img
        alt="user icon"
        src="user-icon.svg"
      />
    </div>
  </div>
  <div
    class="c2"
  >
    <div>
      <h4
        class="c3"
      >
        Location
      </h4>
      <h5
        class="c4"
      >
        Fitzrovia
      </h5>
    </div>
    <button
      class="c5"
      type="button"
    >
      Change Location
    </button>
  </div>
  <p
    class="c6"
  >
    301 Locations
  </p>
</div>
`;
